### DISCUSSION TITLE
Translation in GRID "Does not contain" & DataTime format

### DISCUSSION THREAD

**Discussion by MarekRod** [vikramlearning/blazorbootstrap#904] [Q&A]
![Zrzut ekranu 2024-10-10 090327](https://github.com/user-attachments/assets/26c8d9cb-d72e-4acf-a566-f46c1e61b87f)
Hi, how to translate "Does not contain" in the GRID filter?
Thanks

**Comment by gvreddy04** [comment]
@MarekRod Thanks for using **BlazorBootstrap**. The translation for "Does not contain" is missing in the documentation and demo examples. Please add the following highlighted line to the GridFilterTranslationProvider() method. We will update our docs and demo examples in the next release.

![image](https://github.com/user-attachments/assets/328ebd03-308d-412a-bbd6-48539b5c4dfe)

### C#

```csharp
filtersTranslation.Add(new("!*a*", "Bevat niet", FilterOperator.DoesNotContain));
```

**Reply by MarekRod** [reply]
👍 Thx - works!!

I have one more problem - I can't find anywhere how to change the formatting in the DataTime filter.

I have the formatting .ToString("yyyy-MM-dd HH:mm:ss"), but it doesn't work in the filter :(
How to set it in the control?
![Zrzut ekranu 2024-10-21 223834](https://github.com/user-attachments/assets/c6a69434-ec06-4d8b-8e8d-0d83282c55f0)
